plugins/callback/opentelemetry.py

```diff
@@ -0,0 +1,435 @@
+
+    def set_span_attribute(self, span, attributeName, attributeValue):
+        """ update the span attribute with the given attribute and value if not None """
+
+        if span is None and self._display is not None:
+            self._display.warning('span object is None. Please double check if that is expected.')
+        else:
+            if attributeValue is not None:
+                span.set_attribute(attributeName, attributeValue)
+
+
+class CallbackModule(CallbackBase):
+    """
+    This callback creates distributed traces.
+    """
+
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'notification'
+    CALLBACK_NAME = 'community.general.opentelemetry'
+    CALLBACK_NEEDS_ENABLED = True
+
+    def __init__(self, display=None):
+        super(CallbackModule, self).__init__(display=display)
+        self.hide_task_arguments = None
+        self.otel_service_name = None
+        self.console_output = None
+        self.insecure_otel_exporter = None
+        self.ansible_playbook = None
+        self.play_name = None
+        self.tasks_data = None
+        self.errors = 0
+        self.disabled = False
+
+        if OTEL_LIBRARY_IMPORT_ERROR:
+            raise_from(
+                AnsibleError('The `opentelemetry-api`, `opentelemetry-exporter-otlp` or `opentelemetry-sdk` must be installed to use this plugin'),
+                OTEL_LIBRARY_IMPORT_ERROR)
+
+        if ORDER_LIBRARY_IMPORT_ERROR:
+            raise_from(
+                AnsibleError('The `ordereddict` must be installed to use this plugin'),
+                ORDER_LIBRARY_IMPORT_ERROR)
+        else:
+            self.tasks_data = OrderedDict()
+
+        self.opentelemetry = OpenTelemetrySource(display=self._display)
+
+    def transform_option_to_boolean_or_default(self, option_name, default):
+        """ Look for the given option and if a string boolean then convert to a boolean type """
+
+        return self.transform_to_boolean_or_default(self.get_option(option_name), default)
+
+    def transform_to_boolean_or_default(self, value, default):
+        """ Transform string boolean to boolean or apply default value """
+
+        if value is None:
+            return default
+        if isinstance(value, str):
+            return value.lower() in ['true']
+        return value
+
+    def set_options(self, task_keys=None, var_options=None, direct=None):
+        super(CallbackModule, self).set_options(task_keys=task_keys,
+                                                var_options=var_options,
+                                                direct=direct)
+
+        self.hide_task_arguments = self.transform_option_to_boolean_or_default('hide_task_arguments', False)
+
+        self.otel_service_name = self.get_option('otel_service_name')
+
+        if self.otel_service_name is None:
+            self.otel_service_name = 'ansible'
+
+        self.console_output = self.transform_option_to_boolean_or_default('console_output', False)
```

Comment:
This is more for testing purposes, or dry-run executions rather than sending the traces to the endpoint. It uses https://opentelemetry-python.readthedocs.io/en/stable/sdk/trace.export.html#opentelemetry.sdk.trace.export.ConsoleSpanExporter so I can potentially remove it though I like the idea of having the option to dry-run this.

What do you think?